File: bindings/lua/Polycode/OSFileEntry.lua
class "OSFileEntry"

OSFileEntry.TYPE_FILE = 0
OSFileEntry.TYPE_FOLDER = 1

function OSFileEntry:__getvar(name)
	if name == "name" then
		return Polycode.OSFileEntry_get_name(self.__ptr)
	elseif name == "extension" then
		return Polycode.OSFileEntry_get_extension(self.__ptr)
	elseif name == "nameWithoutExtension" then
		return Polycode.OSFileEntry_get_nameWithoutExtension(self.__ptr)
	elseif name == "basePath" then
		return Polycode.OSFileEntry_get_basePath(self.__ptr)
	elseif name == "fullPath" then
		return Polycode.OSFileEntry_get_fullPath(self.__ptr)
	elseif name == "type" then
		return Polycode.OSFileEntry_get_type(self.__ptr)
	end
end

function OSFileEntry:__setvar(name,value)
	if name == "name" then
		Polycode.OSFileEntry_set_name(self.__ptr, value)
		return true
	elseif name == "extension" then
		Polycode.OSFileEntry_set_extension(self.__ptr, value)
		return true
	elseif name == "nameWithoutExtension" then
		Polycode.OSFileEntry_set_nameWithoutExtension(self.__ptr, value)
		return true
	elseif name == "basePath" then
		Polycode.OSFileEntry_set_basePath(self.__ptr, value)
		return true
	elseif name == "fullPath" then
		Polycode.OSFileEntry_set_fullPath(self.__ptr, value)
		return true
	elseif name == "type" then
		Polycode.OSFileEntry_set_type(self.__ptr, value)
		return true
	end
	return false
end
function OSFileEntry:OSFileEntry(...)
	local arg = {...}
	for k,v in pairs(arg) do
		if type(v) == "table" then
			if v.__ptr ~= nil then
				arg[k] = v.__ptr
			end
		end
	end
	if self.__ptr == nil and arg[1] ~= "__skip_ptr__" then
		self.__ptr = Polycode.OSFileEntry(unpack(arg))
	end
end

function OSFileEntry:init(path, name, type)
	local retVal = Polycode.OSFileEntry_init(self.__ptr, path, name, type)
end

function OSFileEntry:__delete()
	if self then Polycode.delete_OSFileEntry(self.__ptr) end
end

File: bindings/javascript/Polycode/OSFileEntry.js
function OSFileEntry() {
	Object.defineProperties(this, {
		'name': { enumerable: true, configurable: true, get: OSFileEntry.prototype.__get_name, set: OSFileEntry.prototype.__set_name},
		'extension': { enumerable: true, configurable: true, get: OSFileEntry.prototype.__get_extension, set: OSFileEntry.prototype.__set_extension},
		'nameWithoutExtension': { enumerable: true, configurable: true, get: OSFileEntry.prototype.__get_nameWithoutExtension, set: OSFileEntry.prototype.__set_nameWithoutExtension},
		'basePath': { enumerable: true, configurable: true, get: OSFileEntry.prototype.__get_basePath, set: OSFileEntry.prototype.__set_basePath},
		'fullPath': { enumerable: true, configurable: true, get: OSFileEntry.prototype.__get_fullPath, set: OSFileEntry.prototype.__set_fullPath},
		'type': { enumerable: true, configurable: true, get: OSFileEntry.prototype.__get_type, set: OSFileEntry.prototype.__set_type}
	})
}
OSFileEntry.prototype.__get_name = function() {
	return Polycode.OSFileEntry__get_name(this.__ptr)
}

OSFileEntry.prototype.__set_name = function(val) {
	Polycode.OSFileEntry__set_name(this.__ptr, val)
}

OSFileEntry.prototype.__get_extension = function() {
	return Polycode.OSFileEntry__get_extension(this.__ptr)
}

OSFileEntry.prototype.__set_extension = function(val) {
	Polycode.OSFileEntry__set_extension(this.__ptr, val)
}

OSFileEntry.prototype.__get_nameWithoutExtension = function() {
	return Polycode.OSFileEntry__get_nameWithoutExtension(this.__ptr)
}

OSFileEntry.prototype.__set_nameWithoutExtension = function(val) {
	Polycode.OSFileEntry__set_nameWithoutExtension(this.__ptr, val)
}

OSFileEntry.prototype.__get_basePath = function() {
	return Polycode.OSFileEntry__get_basePath(this.__ptr)
}

OSFileEntry.prototype.__set_basePath = function(val) {
	Polycode.OSFileEntry__set_basePath(this.__ptr, val)
}

OSFileEntry.prototype.__get_fullPath = function() {
	return Polycode.OSFileEntry__get_fullPath(this.__ptr)
}

OSFileEntry.prototype.__set_fullPath = function(val) {
	Polycode.OSFileEntry__set_fullPath(this.__ptr, val)
}

OSFileEntry.prototype.__get_type = function() {
	return Polycode.OSFileEntry__get_type(this.__ptr)
}

OSFileEntry.prototype.__set_type = function(val) {
	Polycode.OSFileEntry__set_type(this.__ptr, val)
}

Duktape.fin(OSFileEntry.prototype, function (x) {
	if (x === OSFileEntry.prototype) {
		return;
	}
	Polycode.OSFileEntry__delete(x.__ptr)
})

OSFileEntry.prototype.init = function(path,name,type) {
	Polycode.OSFileEntry_init(this.__ptr, path,name,type)
}
